src/utils.ts

```diff
@@ -1344,3 +1344,56 @@ export async function fileIsAccessible(fileName: string, mode?: number) {
     return false;
   }
 }
+
+/** @internal */
+export class MovingWindow {
+  /** Index of the next slot to be overwritten */
+  private writeIndex: number;
+  length: number;
+  samples: Float64Array;
+
+  constructor(windowSize = 10) {
+    this.samples = new Float64Array(windowSize);
+    this.length = 0;
+    this.writeIndex = 0;
+  }
+
+  addSample(sample: number) {
+    this.samples[this.writeIndex++] = sample;
+    if (this.length < this.samples.length) {
+      this.length++;
+    }
+
+    this.writeIndex %= this.samples.length;
+  }
+
+  min(): number {
+    if (this.length < 2) return 0;
```

Comment:
Is this correct?  Wouldn't the min rtt when there's a single sample be `this.samples[0]`?

Reply:
We're implementing the [CSOT spec](https://github.com/mongodb/specifications/blob/bbb335e60cd7ea1e0f7cd9a9443cb95fc9d3b64d/source/client-side-operations-timeout/client-side-operations-timeout.md#drivers-use-minimum-rtt-to-short-circuit-operations) here, so I can rename the class as Neal recommended to be more explicit about what this class is doing and why it's calculating the minimum this way.

From the linked spec:
> A previous version of this spec used the 90th percentile RTT to short circuit operations that might otherwise fail with a socket timeout. We decided to change this logic to avoid canceling operations that may have a high chance of succeeding and also remove a dependency on t-digest. Instead, drivers use the minimum RTT from the last 10 samples, or 0 until at least 2 samples have been recorded.